cob_obstacle_distance/src/distance_manager.cpp

```diff
@@ -0,0 +1,386 @@
+
+        // Representation of segment_of_interest as specific shape
+        fcl::Box b(0.1, 0.1, 0.1);
+        fcl::Sphere s(0.1);
+        fcl::Cylinder c(0.1, 0.1);
+        tPtrMarkerShapeBase ooi;
+        if(!DistanceManager::getMarkerShape(it->second.shape_type, abs_jnt_pos, ooi))
+        {
+            return;
+        }
+
+        fcl::CollisionObject ooi_co = ooi->getCollisionObject();
+        ooi.reset();
+        fcl::DistanceResult dist_result;
+        bool setDistResult = false;
+        fcl::CollisionObject result_collision_obj = ooi_co;
+        fcl::FCL_REAL lastDist = std::numeric_limits<fcl::FCL_REAL>::max();
+        ROS_INFO_STREAM("Iteration over obstacles for segment of interest: " << frame_of_interest_name);
+        for(ShapesManager::iterator it = this->obstacle_mgr_->begin(); it != this->obstacle_mgr_->end(); ++it)
+        {
+            fcl::CollisionObject collision_obj = (*it)->getCollisionObject();
+            fcl::DistanceResult tmpResult;
+            fcl::DistanceRequest distRequest(true, 1.0, 100.0);
+            fcl::FCL_REAL dist = fcl::distance(&ooi_co, &collision_obj, distRequest, tmpResult);
+            if (dist < lastDist)
+            {
+                setDistResult = true;
+                dist_result = tmpResult;
+                result_collision_obj = collision_obj;
+            }
+        }
+
+        if(setDistResult)
+        {
+            ROS_INFO_STREAM("Minimum distance: " << dist_result.min_distance);
+            fcl::Vec3f t = dist_result.nearest_points[1] + result_collision_obj.getTranslation(); // Translation from "base_link" frame!!!
+            fcl::Quaternion3f q = result_collision_obj.getQuatRotation();
+
+            Eigen::Vector3d obst_vector(t[VEC_X],
+                                       t[VEC_Y],
+                                       t[VEC_Z]);
+
+            ROS_INFO_STREAM("absJntPosition: " << abs_jnt_pos);
+            Eigen::Vector3d dist_vector = abs_jnt_pos - obst_vector;
+            cob_obstacle_distance::ObstacleDistance od_msg;
+            od_msg.distance = dist_result.min_distance;
+            od_msg.distance_vector.x = dist_vector(VEC_X);
+            od_msg.distance_vector.y = dist_vector(VEC_Y);
+            od_msg.distance_vector.z = dist_vector(VEC_Z);
+            od_msg.id.frame_id = frame_of_interest_name;
+            od_msg.id.stamp = ros::Time::now();
+            obstacle_distances.distances.push_back(od_msg);
+        }
+    }
+
+    if(obstacle_distances.distances.size() > 0)
+    {
+        this->obstacle_distances_pub_.publish(obstacle_distances);
+    }
+}
+
+bool DistanceManager::transform()
+{
+    bool success = true;
+    try
+    {
+        tf::StampedTransform cb_transform_bl;
+        tf_listener_.waitForTransform(chain_base_link_, "base_link", ros::Time(0), ros::Duration(0.5));
+        tf_listener_.lookupTransform(chain_base_link_, "base_link", ros::Time(0), cb_transform_bl);
```

Comment:
should we already provide the possibility of being able to configure "base_link"?
E.g. check out the parameter "root_frame" of the frame_tracker!